log-manager/src/test/java/io/airlift/log/TestJsonFormatter.java

```diff
@@ -0,0 +1,70 @@
+package io.airlift.log;
+
+import io.airlift.json.JsonCodec;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.logging.LogRecord;
+
+import static com.google.common.base.Throwables.getStackTraceAsString;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.time.temporal.ChronoUnit.NANOS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestJsonFormatter
+{
+    @Test
+    public void testMinimalJsonErrorLogLine()
+    {
+        JsonRecord originalJsonRecord = new JsonRecord(Instant.now(), Level.DEBUG, "thread-0", "TestLogger", "Test Log Message", new Exception("Test Exception 1"));
+
+        RuntimeException exception = new RuntimeException("Test Exception 2");
+        String minimalJsonErrorLogLine = (new JsonFormatter()).minimalJsonErrorLogLine(originalJsonRecord, exception);
+
+        assertTrue(minimalJsonErrorLogLine.endsWith("\n"), "Log lines should end with newline");
```

Comment:
Use AssertJ for this, which we're using in new tests:
```java
assertThat(minimalJsonErrorLogLine).as("Log lines should end with newline").endsWith("\n");
```
This produces a better error message since it shows the actual input. Generally, avoid using `assertTrue()` with a complex expression.